benchmarks/src/main/kotlin/montecarlo/MonteCarloBasedChannelBenchmark.kt

```diff
@@ -0,0 +1,203 @@
+@file:JvmName("MonteCarloBasedBenchmark")
+
+package montecarlo
+
+import kotlinx.coroutines.runBlocking
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.concurrent.Phaser
+import java.util.concurrent.ThreadLocalRandom
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.pow
+import kotlin.math.sqrt
+import kotlin.random.Random
+
+fun main() {
```

Comment:
Please, add documentation with a detailed description of how the benchmark works